apps/api/src/app/subscribers/usecases/handle-chat-oauth/handle-chat-oauth.usecase.ts

```diff
@@ -0,0 +1,126 @@
+import { Injectable } from '@nestjs/common';
+import axios from 'axios';
+import { CreateSubscriber, CreateSubscriberCommand, decryptCredentials } from '@novu/application-generic';
+
+import { HandleChatOauthCommand } from './handle-chat-oauth.command';
+import {
+  IChannelCredentialsCommand,
+  OAuthHandlerEnum,
+  UpdateSubscriberChannel,
+  UpdateSubscriberChannelCommand,
+} from '../update-subscriber-channel';
+import { ChannelTypeEnum, EnvironmentRepository, IntegrationEntity, IntegrationRepository } from '@novu/dal';
+import { ApiException } from '../../../shared/exceptions/api.exception';
+import { ICredentialsDto } from '@novu/shared';
+
+@Injectable()
+export class HandleChatOauth {
+  readonly SLACK_ACCESS_URL = 'https://slack.com/api/oauth.v2.access';
+  readonly SCRIPT_CLOSE_TAB = '<script>window.close();</script>';
+
+  constructor(
+    private updateSubscriberChannelUsecase: UpdateSubscriberChannel,
+    private integrationRepository: IntegrationRepository,
+    private environmentRepository: EnvironmentRepository,
+    private createSubscriberUsecase: CreateSubscriber
+  ) {}
+
+  async execute(command: HandleChatOauthCommand) {
+    const { credentials: integrationCredentials } = await this.getIntegration(command);
+
+    const webhookUrl = await this.getWebhook(command, integrationCredentials);
+
+    const environment = await this.getEnvironment(command);
+
+    await this.createSubscriber(environment, command, webhookUrl);
+
+    const redirect = integrationCredentials.redirectUrl != null && integrationCredentials.redirectUrl != '';
+
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    return { redirect, action: redirect ? integrationCredentials.redirectUrl! : this.SCRIPT_CLOSE_TAB };
+  }
+
+  private async createSubscriber(environment, command: HandleChatOauthCommand, webhookUrl) {
+    await this.createSubscriberUsecase.execute(
+      CreateSubscriberCommand.create({
+        organizationId: environment._organizationId,
+        environmentId: command.environmentId,
+        subscriberId: command?.subscriberId,
+      })
+    );
+
+    const subscriberCredentials: IChannelCredentialsCommand = { webhookUrl: webhookUrl, channel: command.providerId };
+
+    await this.updateSubscriberChannelUsecase.execute(
+      UpdateSubscriberChannelCommand.create({
+        organizationId: environment._organizationId,
+        environmentId: command.environmentId,
```

Comment:
Could we get rid of the need to use the `organizationId` so that would simplify a lot the code? Also checking `UpdateSubscriberChannel` use case I can see we have no use for `organizationId`: https://github.com/novuhq/novu/blob/f1ab32fef885a5422bde34572ca7d794261c3960/apps/api/src/app/subscribers/usecases/update-subscriber-channel/update-subscriber-channel.command.ts#L20
https://github.com/novuhq/novu/blob/f1ab32fef885a5422bde34572ca7d794261c3960/apps/api/src/app/subscribers/usecases/update-subscriber-channel/update-subscriber-channel.usecase.ts#L10

Reply:
Yeah totally agree here, i would love to remove it, but then i will need to make it optional on all the other commands. the UserSession is not an option because we don't have it.
But but we could get it as a param from the user. i will look into it in the refactoring phase now i need to merge it to the next branch so i could test with dev env and not with localhost and ngrok :(